src/Microsoft.AspNetCore.SignalR.Core/DefaultHubLifetimeManager.cs

```diff
@@ -127,6 +127,29 @@ public override Task InvokeGroupAsync(string groupName, string methodName, objec
             return Task.CompletedTask;
         }
 
+        public override Task InvokeGroupsAsync(IReadOnlyList<string> groupNames, string methodName, object[] args)
+        {
+            // Each task represents the list of tasks for each of the writes within a group
+            var tasks = new List<Task>();
+
+            foreach (string groupName in groupNames)
+            {
+                if (groupName == null)
+                {
+                    throw new ArgumentNullException(nameof(groupName));
+                }
+
+                var group = _groups[groupName];
+                if (group != null)
+                {
+                    var message = CreateInvocationMessage(methodName, args);
```

Comment:
Looks like the Invocation can be created outside the `foreach` loop. That would save some allocations.